app/views.py

```diff
@@ -177,3 +189,107 @@ def get_context_data(self, **kwargs):
         context["total_item"] = total_item
         context["total_quantity"] = total_quantity
         return context
+
+
+class OrderView(generic.ListView):
+    """View function for home page of site."""
+
+    paginate_by = CART_VIEW_PAGINATE
+    context_object_name = "order_items"
+    first_user = User.objects.first()
+    first_profile = Profile.objects.filter(user=first_user).first()
+    order = Order.objects.filter(user=first_profile).first()
+
+    def get_queryset(self):
+        order_id = self.kwargs["order_id"]
+        if self.first_user:
+            return OrderItem.objects.filter(order__order_id=order_id)
+        else:
+            return OrderItem.objects.none()
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        order_items = self.get_queryset()
+
+        context["order_id"] = self.kwargs["order_id"]
+        context["user_name"] = self.first_profile.name
+
+        context["total_price"] = self.order.total_price
+        context["total_price_include_shipping"] = self.order.total_price + 5
+        return context
+
+
+def create_order(request):
+    if request.method == "POST":
+        try:
+            first_user = User.objects.first()
+            cart = Cart.objects.filter(user=first_user).first()
+            cart_items = CartItem.objects.filter(cart=cart)
+            if not cart or not cart_items:
+                return JsonResponse({"error": "Cart is not found"}, status=400)
+
+            profile = Profile.objects.get(user=first_user)
+
+            max_payment_id = Payment.objects.aggregate(
+                models.Max("payment_id")
+            )["payment_id__max"]
+            next_payment_id = int(max_payment_id) + 1 if max_payment_id else 1
+            payment = Payment.objects.create(
+                payment_id=str(next_payment_id),
+                payment_date=timezone.now(),
+                amount=0,
+                method="Credit Card",
+            )
+
+            restaurant = cart_items.first().item.restaurant
+            if not restaurant:
+                return JsonResponse(
+                    {"error": "Restaurant not found"}, status=400
+                )
+
+            order = Order.objects.create(
+                user=profile,
+                restaurant=restaurant,
+                total_price=0,
+                status="Pending",
+                payment=payment,
+            )
+
+            total_price = 0
+            order_items = []
+
+            for cart_item in cart_items:
+                item_price = cart_item.item.price * cart_item.quantity
+                total_price += item_price
+
+                order_item = OrderItem(
+                    order=order,
+                    item=cart_item.item,
+                    quantity=cart_item.quantity,
```

Comment:
trong DB có quản lý số lượng sp ko em? nếu cõ chỗ này cần thêm xử lý validate nếu số lược item order > số lượng còn lại của item trong Db em nhé

Reply:
Trường hợp item order > số lượng còn lại của item em đã xử lý trong cart rồi ạ. Người dùng chỉ có thể tạo order từ cart thôi nên trong trường hợp này thì sẽ không xảy ra lỗi ạ

Reply:
Vậy sẽ có trg hợp khi User 1 add sp A vào giỏ hàng thành công, User 2 cũng add sp A vào giỏ hàng thành công và tiến hành thanh toán trc (sp A hết hàng) => thì User 1 có thanh toán thành công ko em??

Reply:
Vâng ạ, đúng là có trường hợp này. Em đã thêm logic để xử lý trường hợp này ạ.